File: ios-cuantrack/CuanTrack/Views/Components/CategoryChip.swift
//
//  CategoryChip.swift
//  CuanTrack
//

import SwiftUI

struct CategoryChip: View {
    let category: TransactionCategory
    let isSelected: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: 6) {
                Image(systemName: category.icon)
                    .font(.system(size: 13))

                Text(category.rawValue)
                    .font(.system(size: 14, weight: .medium))
            }
            .padding(.horizontal, 14)
            .padding(.vertical, 10)
            .background(isSelected ? category.color : Color(red: 0.95, green: 0.96, blue: 0.97))
            .foregroundStyle(isSelected ? .white : Color(red: 0.39, green: 0.45, blue: 0.55))
            .clipShape(.capsule)
        }
        .buttonStyle(.plain)
    }
}

#Preview {
    HStack(spacing: 8) {
        CategoryChip(category: .food, isSelected: true) {}
        CategoryChip(category: .transport, isSelected: false) {}
        CategoryChip(category: .shopping, isSelected: false) {}
    }
    .padding(24)
}

File: android-cuantrack/app/src/main/java/com/rork/cuantrackandroid/ui/components/CategoryChip.kt
package com.rork.cuantrackandroid.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Restaurant
import androidx.compose.material.icons.filled.DirectionsCar
import androidx.compose.material.icons.filled.ShoppingBag
import androidx.compose.material.icons.filled.Description
import androidx.compose.material.icons.filled.Tv
import androidx.compose.material.icons.filled.MoreHoriz
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.rork.cuantrackandroid.models.TransactionCategory

@Composable
fun CategoryChip(
    category: TransactionCategory,
    isSelected: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .clip(RoundedCornerShape(20.dp))
            .background(if (isSelected) category.color else Color(0xFFF1F5F9))
            .clickable(onClick = onClick)
            .padding(horizontal = 14.dp, vertical = 10.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = category.iconVector(),
            contentDescription = null,
            tint = if (isSelected) Color.White else Color(0xFF64748B),
            modifier = Modifier.size(14.dp)
        )
        Spacer(Modifier.size(6.dp))
        Text(
            category.label,
            fontSize = 14.sp,
            fontWeight = FontWeight.Medium,
            color = if (isSelected) Color.White else Color(0xFF64748B)
        )
    }
}

fun TransactionCategory.iconVector(): ImageVector = when (this) {
    TransactionCategory.FOOD -> Icons.Filled.Restaurant
    TransactionCategory.TRANSPORT -> Icons.Filled.DirectionsCar
    TransactionCategory.SHOPPING -> Icons.Filled.ShoppingBag
    TransactionCategory.BILLS -> Icons.Filled.Description
    TransactionCategory.ENTERTAINMENT -> Icons.Filled.Tv
    TransactionCategory.OTHER -> Icons.Filled.MoreHoriz
}
